addons/account/models/chart_template.py

```diff
@@ -986,19 +1041,64 @@ def _load_translations(self, langs=None, companies=None):
         companies = companies or self.env['res.company'].search([('chart_template', 'in', available_template_codes)])
 
         translation_importer = TranslationImporter(self.env.cr, verbose=False)
+
+        # translate records that are created from the chart_template data
+
         for chart_template, chart_companies in groupby(companies, lambda c: c.chart_template):
             template_data = self.env['account.chart.template']._get_chart_template_data(chart_template)
             template_data.pop('template_data', None)
             for mname, data in template_data.items():
                 for _xml_id, record in data.items():
+                    _name_code_translation_module = record.pop('_name_code_translation_module', 'account')
                     fnames = {fname.split('@')[0] for fname in record}
                     for lang in langs:
                         for fname in fnames:
+                            if not self.env[mname]._fields[fname].translate:
+                                continue
                             value = record.get(f"{fname}@{lang}")
                             if not value:  # manage generic locale (i.e. `fr` instead of `fr_BE`)
                                 value = record.get(f"{fname}@{lang.split('_')[0]}")
+                            if not value and fname == 'name':
+                                value = code_translations.get_python_translations(_name_code_translation_module, lang).get(record[fname])
+                                if not value:  # manage generic locale (i.e. `fr` instead of `fr_BE`)
+                                    value = code_translations.get_python_translations(_name_code_translation_module, lang.split('_')[0]).get(record[fname])
                             if value:
                                 for company in chart_companies:
                                     xml_id = f"account.{company.id}_{_xml_id}"
                                     translation_importer.model_translations[mname][fname][xml_id][lang] = value
+
+        # translate the name of TEMPLATE_MODELS records that are not created from the chart_template data
+
+        # We use all modules that modify the model of self ('account.chart.template') as translation source.
```

Comment:
It's not linked to `account.chart.template`, is it?

And for these values, isn't it always in `account`? Do you have an example for data that comes from another module but that is not declared with `@template`?

Reply:
It was mainly for the demo data.

I guess anything that should be part of the chart data could just be defined in an `@template` function anyway?

I only managed to find this 1 example that is directly linked to the chart template.
(But the names are not translated anyway currently and a bit technical maybe.)
https://github.com/odoo/enterprise/blob/fb85e33cb8853aa72ec3ac66f3e58d06c325fec2/l10n_ec_edi/models/account_chart_template.py